web/src/Modal.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { Component, ChangeEvent, FormEvent } from 'react';
+
+class Modal extends Component<any> {
+  state: {
+    Num1: any,
+    Num2: any,
+    I: any,
+  }
+
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      Num1: '',
+      Num2: '',
+      I: '',
+    }
+
+    this.createRow = this.createRow.bind(this);
+    this.cancelCreating = this.cancelCreating.bind(this);
+    this.changeNum1 = this.changeNum1.bind(this);
+    this.changeNum2 = this.changeNum2.bind(this);
+    this.changeI = this.changeI.bind(this);
+  }
+
+  createRow(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    const { Num1, Num2, I } = this.state;
+
+    const data = {
+      Num1: +Num1,
+      Num2: +Num2,
+      I: +I,
+    }
+
+    this.props.request(data);
+    this.props.closeModal();
+  }
+
+  cancelCreating() {
+    this.props.closeModal();
+  }
+
+  changeNum1(event: ChangeEvent<HTMLInputElement>) {
```

Comment:
Don't think this moment the best decision, but best what i know, [react documentation](https://reactjs.org/docs/forms.html) don't have input multiply checks. In future i will replace that.